floor_generator/lib/misc/extension/string_extension.dart

```diff
@@ -16,4 +16,40 @@ extension StringExtension on String {
         return this[0].toLowerCase() + substring(1);
     }
   }
+
+  /// Returns a copy of this string having its first letter uppercased, or the
+  /// original string, if it's empty or already starts with a upper case letter.
+  ///
+  /// ```dart
+  /// print('abcd'.capitalize()) // Abcd
+  /// print('Abcd'.capitalize()) // Abcd
+  /// ```
+  String capitalize() {
+    switch (length) {
+      case 0:
+        return this;
+      case 1:
+        return toUpperCase();
+      default:
+        return this[0].toUpperCase() + substring(1);
+    }
+  }
+}
+
+extension NullableStringExtension on String? {
+  /// Converts this string to a literal for
+  /// embedding it into source code strings.
+  ///
+  /// ```dart
+  /// print(null.toLiteral())   // null
+  /// print('Abcd'.toLiteral()) // 'Abcd'
+  /// ```
+  String toLiteral() {
```

Comment:
it is currently used for writing out the query parts of the query method body. It should probably also used for the query part of the views. Having looked at that just now, I should change the current "escaping algorithm" a small bit (use `"''' '''"` instead of `"' '"`). This still leaves some holes but they should be rarely used.

Reply:
scratch that, changing this will blow up this PR rapidly (many tests have to be changed). I think this can be addressed better in a separate PR.